oshi-core/src/main/java/oshi/util/ProcUtil.java

```diff
@@ -0,0 +1,124 @@
+     *         "IpExt": {"InNoRoutes":55, "InTruncatedPkts":27786, "InMcastPkts":1435, ... }
+     *         "MPTcpExt": {"MPCapableSYNACKRX":0, "MPCapableSYNTX":0, "MPCapableSYNACKRX":0, ... }
+     *     }
+     * </pre>
+     *
+     * @param procFile the file to process
+     * @param keys an optional array of keys to return in the outer map. If none are given, all
+     *             found keys are returned.
+     * @return a map of keys to stats
+     */
+    public static Map<String, Map<String, Long>> parseNestedStatistics(String procFile, String... keys) {
+        Map<String, Map<String, Long>> result = new HashMap<>();
+        Set<String> keysSet = new HashSet<>(Arrays.asList(keys));
+
+        List<String> lines = FileUtil.readFile(procFile);
+        String previousKey = null;
+        String[] statNames = null;
+
+        for (String line : lines) {
+            String[] parts = ParseUtil.whitespaces.split(line);
+            if (parts.length == 0) {
+                continue;
+            }
+            String key = parts[0].substring(0, parts[0].length() - 1);
+
+            if (!keysSet.isEmpty() && !keysSet.contains(key)) {
+                continue;
+            }
+
+            if (key.equals(previousKey)) {
+                if (parts.length == statNames.length) {
+                    Map<String, Long> stats = new HashMap<>(parts.length - 1);
+                    for (int i = 1; i < parts.length; i++) {
+                        stats.put(statNames[i], ParseUtil.parseLongOrDefault(parts[i], 0));
+                    }
+                    result.put(key, stats);
+                }
+            } else {
+                statNames = parts;
+            }
+
+            previousKey = key;
+        }
+
+        return result;
+    }
+
+    /**
+     * Parses /proc files formatted as "statistic  (long)value" to produce a simple mapping. An
+     * example would be /proc/net/snmp6. The file format would look like:
+     * <pre>
+     *    Ip6InReceives             8026
+     *    Ip6InHdrErrors            0
+     *    Icmp6InMsgs               2
+     *    Icmp6InErrors             0
+     *    Icmp6OutMsgs              424
+     *    Udp6IgnoredMulti          5
+     *    Udp6MemErrors             1
+     *    UdpLite6InDatagrams       37
+     *    UdpLite6NoPorts           1
+     * </pre>
+     * Which would produce a mapping structure like:
+     * <pre>
+     *     {
+     *         "Ip6InReceives":8026,
+     *         "Ip6InHdrErrors":0,
+     *         "Icmp6InMsgs":2,
+     *         "Icmp6InErrors":0,
+     *         ...
+     *     }
+     * </pre>
+     *
+     * @param procFile the file to process
+     * @param separator a regex specifying the separator between statistic and value. For
+     *                  whitespace use {@code "\\s+"}.
+     * @return a map statistics and associated values
+     */
+    public static Map<String, Long> parseStatistics(String procFile, String separator) {
```

Comment:
Let's make `separator` a `Pattern`, so line 116 can be `separator.split(line)`.  Otherwise you're doing regex splits that compile each time inside a loop.  Would be nice to make it an optional argument (overloaded method) with a `ParseUtil.whitespaces` default.

Reply:
Done - added the suggested overloaded method.